### PR TITLE
[Backport]-issue-195196 Customer related values are NULL for guests converted to customers after checkout.

### PR DESCRIPTION
**Pull request by Nazar65**
### Description (*)
Customer name and all customer related values should not be NULL

### Fixed Issues (if relevant)
1. magento/magento2#19166:Customer related values are NULL for guests converted to customers after checkout.
### Manual testing scenarios (*)

1. Allow customers to create account after checkout
2. Create few orders as guest customers and create an account after checkout process
3. Create a module with a plugin aroundSave for SalesOrder class
4. In aroundSave method, try to fetch customer name from order object
```
public function aroundSave(
        OrderResource $subject,
        \Closure $proceed,
        SalesOrder $order
    ) {
        [...]
        if ($order->getCustomerId()) {
            $customerName = $order->getCustomerName(); // returns Customer Name
        [...]
        }
    }
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by Nazar65** [comment]
HI @rogyar any updates ?

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 1]
```diff
@@ -355,6 +356,13 @@ public function placeOrder($cartId, PaymentInterface $paymentMethod = null)
         if ($quote->getCheckoutMethod() === self::METHOD_GUEST) {
             $quote->setCustomerId(null);
             $quote->setCustomerEmail($quote->getBillingAddress()->getEmail());
+            if ($quote->getCustomerFirstname() === null && $quote->getCustomerLastname() === null) {
+                $quote->setCustomerFirstname($quote->getBillingAddress()->getFirstname());
+                $quote->setCustomerLastname($quote->getBillingAddress()->getLastname());
+                if ($quote->getBillingAddress()->getMiddlename() === null) {
```
Comment (sivaschenko) [inline app/code/Magento/Quote/Model/QuoteManagement.php]: Is this check by design, or did you mean to check `$quote->hasCustomerMiddlename` here?

**Review by sivaschenko — CHANGES_REQUESTED** [review]
hi @Nazar65 thanks for backport, I have a question regarding the implementation, can you please check my code review note

File: app/code/Magento/Quote/Model/QuoteManagement.php (hunk 1 above)
Comment (Nazar65) [reply app/code/Magento/Quote/Model/QuoteManagement.php]: Yes in original Pr i was check if `$quote->hasCustomerMiddlename` but now i don't know how i did this, i'm will add a propper check

**Comment by Nazar65** [comment]
Hi @sivaschenko i've added a valid check for customer middle name, is this check be valid ? or we can add a method as you mentioned` hasCustomerMiddlename` ?

**Review by sivaschenko — APPROVED** [review]
Thanks @Nazar65 , no need to add new methods

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-4390](https://jira.corp.magento.com/browse/ENGCOM-4390) has been created to process this Pull Request

**Pull request merged**